DocGen4/Output/DocString.lean

```diff
@@ -131,22 +151,47 @@ def addHeadingAttributes (el : Element) (modifyElement : Element → HtmlM Eleme
     let newAttrs := attrs
       |>.insert "id" id
       |>.insert "class" "markdown-heading"
-    let newContents := (←
-      contents.mapM (fun c => match c with
-      | Content.Element e => return Content.Element (← modifyElement e)
-      | _ => pure c))
+    let newContents ← modifyContents contents funName modifyElement
+    return ⟨ name, newAttrs, newContents
       |>.push (Content.Character " ")
-      |>.push (Content.Element anchor)
-    return ⟨ name, newAttrs, newContents⟩
+      |>.push (Content.Element anchor) ⟩
 
 /-- Extend anchor links. -/
-def extendAnchor (el : Element) : HtmlM Element := do
+def extendAnchor (el : Element) (funName : String) :
+    ModuleToHtmlM Element := do
   match el with
   | Element.Element name attrs contents =>
-    let newAttrs ← match attrs.find? "href" with
-    | some href => pure (attrs.insert "href" (← extendLink href))
-    | none => pure attrs
-    return ⟨ name, newAttrs, contents⟩
+    match attrs.find? "href" with
+    | some href =>
+      let refsMap := (← read).refsMap
```

Comment:
Please introduce an explicit monadic helper function that does this `read` + `find?` pattern on its own. This will make it easier to refactor things in the future.

Reply:
Done.